### ISSUE TITLE
Slug with Arabic letters

### ISSUE THREAD

**Issue by RJFares** [spatie/laravel-sluggable#188]
The slug method expects a string or null as $language (@param  string|null  $language) . In my case, I need the null because I need the Arabic letters NOT to be converted to Latin. 

In this library, SlugOptions.php

public string $slugLanguage = 'en';

This is forcing me to not use null, and there was no way I can get a slug with non-latin letters.

Maybe the only solution for such a case is to instead have:

public **?string** $slugLanguage = 'en';

AND

usingLanguage(~~string~~ $language)

That way, the function responsible for translating to latin letters will be bypassed in Support/Str.php slug(....)

And I can successfully call SlugOptions::createWithLocales($keys)->usingLanguage(null) and get the correct behavior.

**Comment by RJFares** [comment]
Using '' does not solve the issue because the ascii method will return an empty string.

Slugging with non latin letters should be available since if we target another language, the SEO is improved a lot if the title has the correct alphabet and not just and ASCII representation in Latin.

That's why I suggest using ?string as stated in the previous comment.

**Comment by RJFares** [comment]
But this causes an issue where users may want to ASCII some locales and not others...
Any ideas?

**Comment by ZaidBarghouthi** [comment]
See in `Spatie\Sluggable\HasSlug` trait

```php
protected function generateNonUniqueSlug(): string
{
    $slugField = $this->slugOptions->slugField;

    if ($this->hasCustomSlugBeenUsed() && ! empty($this->$slugField)) {
        return $this->$slugField;
    }

    return Str::slug($this->getSlugSourceString(), $this->slugOptions->slugSeparator, $this->slugOptions->slugLanguage);
}
```

The slug is generated using the `Str::slug`. To implement custom character mapping will need a significant change to the code here. 

I'm working on a project at the moment where I need to allow for Arabic in the slugs. I'm just using `->usingLanguage('')`. Can you explain the danger in that?

**Comment by RJFares** [comment]
@ZaidBarghouthi for your project if it's only en/ar it's fine.
I found the method ->usingLanguage limiting because it accepts only a string and not an array, so if I want to have:

en->en, ar->null (or '') and nl->nl I couldn't be that specific so I just reverted to my own snippet.
I did a quick test now on '' again and it worked correctly (not sure why I got empty strings last night!)

I'll close the issue because I can't reproduce it.
